xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp

```diff
@@ -1083,9 +1085,7 @@ void CRenderManager::PrepareNextRender()
   double frameOnScreen = m_dvdClock.GetClock();
   double frametime = 1.0 / g_graphicsContext.GetFPS() * DVD_TIME_BASE;
 
-  // correct display latency
-  // internal buffers of driver, assume that driver lets us go one frame in advance
-  double totalLatency = DVD_SEC_TO_TIME(m_displayLatency) - DVD_MSEC_TO_TIME(m_videoDelay) + 2* frametime;
+  double totalLatency = DVD_SEC_TO_TIME(TotalLatency());
```

Comment:
> so you cannot render faster than vrefresh in that case

but we get one frame ahead. assume we have not called swapBuffers for a while. now the first call to swapBuffers will not block because of double buffers.

> Should I take the time from PresentRender at function exit or enter, i.e. before or after eglSwapBuffers (makes a big difference)?

if swapBuffers blocks, it makes a difference.

> And is this similar to https://github.com/FernetMenta/kodi-agile/blob/master/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp#L1097 or something else altogether?

this code gets the time since last vblank. not sure if I get your question